internal/util/method.go

```diff
@@ -182,11 +182,11 @@ type MethodDelete struct {
 	Resource *annotations.ResourceDescriptor
 	Method   *protogen.Method
 
-	Name string
+	UserProvidedName string
```

Comment:
Perhaps I'm missing something, but I dont see where this upcoming refactor is and what it makes easier?
Consider reverting this as Name should be unambiguous in this context.

Reply:
Ah, now I see it in commit for required etag!
Is it possible to remove the `Name` (or `UserProvidedName` after rename) completely and just use `ResourceVar`? That would be more clear and require less conditionals in the method generators.

Reply:
I think its hard to get rid of `UserProvidedName` and `UserProvidedEtag` since they are used when you want to set a explicit value, and that take precedence `ResourceVar.Name|Etag`.
`UserProvidedName` are for example assign when you want to test to send in a malform name or a missing resource.
`UserProvidedEtag` are set when you want to send in a old etag or etag from a previous created resource.

The idea is that `MethodDelete.Generate` takes care of assigning etag if its required or explicit assigned.

But I'm open if we can find a better way to support all the difference cases :sweat_smile:

Reply:
I did a second attempt, got rid of some quirks, please check https://github.com/einride/protoc-gen-go-aip-test/pull/244